File: 00/0005/0005.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int l, n;
    cin >> l >> n;
    vector<int> w(n);
    for (int i = 0; i < n; i++) cin >> w[i];
    sort(w.begin(), w.end());
    int ans = 0, s = 0;
    for (int i = 0; i < n; i++) {
        s += w[i];
        if (s > l) break;
        ans++;
    }
    cout << ans << '\n';
}

File: 00/0005/0005.py
l = int(input())
n = int(input())
w = list(map(int, input().split()))
w.sort()
ans = 0
s = 0
for i in range(n):
	s += w[i]
	if s <= l:
		ans += 1
		continue
	break
print(ans)